docs/api/io/BufferedSerial.md

```diff
@@ -0,0 +1,73 @@
+# BufferedSerial
+
+<span class="images">![](https://os.mbed.com/docs/development/mbed-os-api-doxy/classmbed_1_1_buffered_serial.png)<span>`BufferedSerial` class hierarchy</span></span>
+
+The `BufferedSerial` class provides UART functionality. We recommend you use this class for serial data transfers. You can use it to send and receive bytes of data in a sequence using separate transmit (TX) and receive pins (RX). Communication can be between two processors or for sending text to a console.
+
+Serial channels have the following characteristics:
+
+- TX and RX pins - can be specified as Not Connected (NC) for simplex (unidirectional) communication or as valid pins for full duplex (bidirectional) communication.
+- Baud rate - predefined speed at which data is sent and received. Standard baud rates include 9600, 119200 and 115200.
+
+Data is transmitted using packets of configurable sizes divided in different sections, which include:
+
+- Start bit: indicates the start of UART data transmission.
+- Data frame: can be 5 to 8 (or 9 if a parity bit is not used) bits for the actual data being transferred.
+- Parity bit: optional bit, used for data error detection.
+- Stop bits: 1-2 bits to signal the end of a data packet.
+
+The `BufferedSerial` calls the underlying HAL API functions. Please see the [porting guide](../porting/serial-port.html) for target serial support.
+
+When the RX interrupt is trigged, the `BufferedSerial` class stores the byte(s) available to read from the hardware buffer to an internal intermediary buffer. When a read request is made, the `BufferedSerial` class uses a mutex lock and enters a critical section to read out the number of bytes requested if as many are available in the intermediary buffer.
```

Comment:
What do we mean by "the RX is trigged," and who or what trigged it?
Who or what makes the read request?

Reply:
`RX` is explained in the first paragraph of the page. It is triggered when receiving data from whatever is connected to that interface (a sensor, a console, etc...).